go.mod

```diff
@@ -51,7 +51,7 @@ require (
 	github.com/multiformats/go-multiaddr-dns v0.3.1
 	github.com/multiformats/go-multihash v0.2.3
 	github.com/onflow/atree v0.6.1-0.20230711151834-86040b30171f
-	github.com/onflow/cadence v1.0.0-M3
+	github.com/onflow/cadence v1.0.0-preview.2.0.20240125192801-40b4d6928fcf
```

Comment:
https://github.com/onflow/cadence/pull/3043 got merged after the `v1.0.0-M3` tag. So bumped the cadence version to the commit 40b4d6928fcfcd334a62dc934549ff32a5b9f0f7

Reply:
In an unrelated PR I'd also want https://github.com/onflow/cadence/pull/3042. 

Let's just create a new tag and update to that one